app/models/deploy.rb

```diff
@@ -90,6 +92,24 @@ def variables
     stack.deploy_variables
   end
 
+  def refute!
+    return if failed?
+    flap! unless flapping?
+    update!(confirmations: [confirmations - 1, -1].min)
+    failure! if confirmed?
+  end
+
+  def confirm!
```

Comment:
Yeah I'm not a huge fan of that naming either. I'll try to find a better terminology.